src/highlight/highlight.ts

```diff
@@ -22,15 +27,22 @@ export default class Highlight {
     this.agents = agents;
   }
 
-  postJoint(params: PostJointRequest) {
-    return this.asyncLock.acquire('postJoint', () => this._postJoint(params));
+  postMessage(request: PostMessageRequest) {
+    return this.asyncLock.acquire('postMessage', () =>
+      this._postMessage(request)
+    );
   }
 
-  async _postJoint(params: PostJointRequest) {
+  async _postMessage(request: PostMessageRequest) {
     let steps = 0;
 
+    const salt = await this.adapter.get(`salts:${request.domain.salt}`);
```

Comment:
Since we have salt verification here, we dont need to have it at the agent level